src/code_gen/gen_c_code.jl

```diff
@@ -589,22 +626,22 @@ function gen_main(lang::LangC, T, fname, funname; A = 10::Union{Integer,Matrix})
         if isa(A, Matrix)
             n = LinearAlgebra.checksquare(A)
             push_code!(code, "size_t n = $n;")
-            push_code!(code, "blas_type A[$(n*n)] = {")
+            push_code!(code, "blas_type A[$(n * n)] = {")
             print_indented_matrix(lang, code, A, ind_lvl = 2)
             push_code!(code, "};")
         else # A is an integer
             n = A
             push_code!(code, "size_t n = $n;")
             push_code!(code, "srand(0);")
-            push_code!(code, "blas_type *A = malloc(n*n*sizeof(*A));")
-            push_code!(code, "for(i=0; i<n*n; i++){")
-            push_code!(code, "A[i] = rand() / (1.0*RAND_MAX);", ind_lvl = 2)
+            push_code!(code, "blas_type *A = malloc(n * n * sizeof(*A));")
+            push_code!(code, "for(i = 0; i < n * n; i++){")
+            push_code!(code, "A[i] = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
```

Comment:
separate between complex and real cases? 
```
if eltype(g) <: Complex 
            push_code!(code, "A[i].real = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
            push_code!(code, "A[i].imag = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
else
            push_code!(code, "A[i] = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
end
```

Reply:
> separate between complex and real cases?
> 
> ```
> if eltype(g) <: Complex 
>             push_code!(code, "A[i].real = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
>             push_code!(code, "A[i].imag = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
> else
>             push_code!(code, "A[i] = rand() / (1.0 * RAND_MAX);", ind_lvl = 2)
> end
> ```

Had to do this in a slightly roundabout way because complex types are not assigned in the same way in MKL and OpenBLAS, but this is the idea I implemented.